src/main/java/core/basesyntax/operation/SupplyHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class SupplyHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.SUPPLY) {
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+            shopServiceImpl.addFruits(fruit, quantity);
+        } else {
+            throw new RuntimeException("Unsupported operation for SupplyHandler");
+        }

```

Comment:
It's not a good practice to throw an exception when the operation is not supported. Consider removing this check. The class `OperationStrategyImpl` should be responsible for choosing the right handler for each operation.